File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/TellDonotAsk/Good/Unit.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TellDoNotAsk.Good
{
    public class Unit
    {
        private double _value { get; set; }
        private UnitType _unitType { get; set; }
        public Unit(double value, UnitType unitType)
        {
            _value = value;
            _unitType =  unitType;
        }

        public Unit ConvertTo(UnitType unitType)
        {
            //check the source unit type
            //convert to destination unit type

            //dummy return
            return new Unit(10, unitType);
        }
    }
}

File: TDD Boot Camp/Part 01/Code/OOPrinciples/src/com/oop/telldonotask/good/Unit.java
package com.oop.telldonotask.good;

public class Unit {
	private double value;
	private UnitType unitType;
	public Unit(double value,UnitType unitType) {
		this.value = value;
		this.setUnitType(unitType);
	}
	public double getValue() {
		return value;
	}

	public void setValue(double value) {
		this.value = value;
	}
	public UnitType getUnitType() {
		return unitType;
	}
	public void setUnitType(UnitType unitType) {
		this.unitType = unitType;
	}
	
	public Unit convertTo(UnitType unitType) {
		//check the source unit type
		//convert to destination unit type
		
		//dummy return
		return new Unit(10,unitType);
	}
	
}
